### PR TITLE
Convert Setting

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Convert `Setting` to Swift.
To prevent repeated getter and setter of `UserDefaults`, `subscript` method with generic is created. The example of accessing a default setting value:

set:
```
Settings.shared[.sendButtonDisabled] = true
///set a enum 
Settings.shared[.preferredCamera] = currentCamera
```

get:
```
let disableSendButton: Bool? = Settings.shared[.sendButtonDisabled]
///get a enum 
let settingsCamera: SettingsCamera? = Settings.shared[.preferredCamera]
```

### CODE REVIEW

File: Wire-iOS/Settings+ColorScheme.swift [hunk 1]
```diff
@@ -33,33 +33,23 @@ enum SettingsColorScheme {
...
-        set {            
-            defaults.set(string(for: colorScheme), forKey: UserDefaultColorScheme)
-            defaults.synchronize()
-            notifyColorSchemeChanged()
-        }
+        return .light
     }
-    
+
+    ///TODO: move to SettingsColorScheme, as a init method
```
Comment (billypchan) [inline Wire-iOS/Settings+ColorScheme.swift]: Will be done in seperated PR.

File: Wire-iOS/Settings+ColorScheme.swift [hunk 2]
```diff
@@ -33,33 +33,23 @@ enum SettingsColorScheme {
...
         return UserDefaults.standard
     }
 
-    var colorScheme: SettingsColorScheme {
-        get {
-            guard let string = defaults.string(forKey: UserDefaultColorScheme) else { return .light }
-            
-            return settingsColorScheme(from: string)
+    var colorSchemeVariant: ColorSchemeVariant {
+        if let string: String = self[.colorScheme] {
```
Comment (typfel) [inline Wire-iOS/Settings+ColorScheme.swift]: Still thinks guard makes sense here.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift [hunk 3]
```diff
@@ -156,7 +156,7 @@ extension ConversationContentViewController: UIAdaptivePresentationControllerDel
         keyboardAvoiding.preferredContentSize = CGSize.IPadPopover.preferredContentSize
         keyboardAvoiding.modalPresentationStyle = .popover
 
-        let presenter: PopoverPresenterViewController? = (self.presentedViewController ?? UIApplication.shared.keyWindow?.rootViewController) as? PopoverPresenterViewController
+        let presenter: PopoverPresenterViewController = (presentedViewController ?? UIApplication.shared.keyWindow?.rootViewController) as! PopoverPresenterViewController
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift]: Why adding force unwrapping?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+MessageDeletion.swift [hunk 4]
```diff
@@ -61,7 +61,7 @@ extension CollectionCell: SelectableView {
 
 final class DeletionDialogPresenter: NSObject {
 
-    private weak var sourceViewController: UIViewController?
+    private unowned let sourceViewController: PopoverPresenterViewController
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+MessageDeletion.swift]: We are sure that the `sourceViewController` never goes away?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift]: fixed the merge issue.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+MessageDeletion.swift (hunk 4 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+MessageDeletion.swift]: undoed this line for merge issue.

File: Wire-iOS/Settings+ColorScheme.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Settings+ColorScheme.swift]: refactored the if to guard.

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 5]
```diff
@@ -168,7 +168,7 @@ final class AppRootViewController: UIViewController {
             self.sessionManager?.switchingDelegate = self
             self.sessionManager?.urlActionDelegate = self
             sessionManager.updateCallNotificationStyleFromSettings()
-            sessionManager.useConstantBitRateAudio = Settings.shared().callingConstantBitRate
+                sessionManager.useConstantBitRateAudio = Settings.shared[.callingConstantBitRate] ?? false
```
Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: Wrong indentation

File: Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift [hunk 6]
```diff
@@ -63,20 +63,19 @@ class SettingsPropertyFactory {
...
+        SettingsPropertyName.messageSoundName               : .messageSoundName,
+        SettingsPropertyName.callSoundName                  : .callSoundName,
+        SettingsPropertyName.pingSoundName                  : .pingSoundName,
+        SettingsPropertyName.disableSendButton              : .sendButtonDisabled,
+        SettingsPropertyName.mapsOpeningOption              : .mapsOpeningRawValue,
+        SettingsPropertyName.browserOpeningOption           : .browserOpeningRawValue,
+        SettingsPropertyName.tweetOpeningOption             : .twitterOpeningRawValue,
+        SettingsPropertyName.callingProtocolStrategy        : .callingProtocolStrategy,
+        SettingsPropertyName.enableBatchCollections         : .enableBatchCollections,
+        SettingsPropertyName.callingConstantBitRate         : .callingConstantBitRate,
```
Comment (typfel) [inline Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift]: Why is the mapping for `SettingsPropertyName.disableLinkPreviews`removed?

Comment (billypchan) [reply Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift]: Checked that `disableLinkPreviews` is a wrapper of `ExtensionSettings.shared.disableLinkPreviews` and not saved to UserDefault. The mapping is not needed.

**Pull request merged**